src/styles/mixins.scss

```
@@ -0,0 +1,78 @@
+@mixin hover ($property, $toValue) {
+  transition: #{$property} $effect-duration;
+  &:hover {
+    #{$property}: #{$toValue};
+  }
+}
+
+@mixin colorHover ($property, $toValue) {
+  transition: #{$property} $effect-duration;
+  &:hover {
+    #{$property}: #{$toValue};
+  }
+}
+
+@mixin largeAndChangeColorOnHover {
+  &:hover {
+    transform: scale(1.2);
+    z-index: 1;
+    transition: transform $effect-duration,
+      color $effect-duration;
+  }
+}
+
+@mixin onDesctop {
+  @media (min-width: $desctop-min-width) {
+    @content;
+  }
+}
+
+@mixin onTablet {
+  @media (min-width: $tablet-min-width) {
+    @content;
+  }
+}
+
+@mixin onPhone {
+  @media (min-width: $phone-min-width) {
+    @content;
+  }
+}
+
+@mixin grid {
+  --columns: 1;
+  display: grid;
+  grid-template-columns: repeat(var(--columns),$card-width);
+
+  @include onPhone {
+    --columns: 2;
+    column-gap: $column-gap;
+    row-gap: $row-gap;
+  }
+
+  @include onTablet {
+    --columns: 3;
+    column-gap: $column-gap;
+    row-gap: $row-gap;
+  }
+
+  @include onDesctop {
+    --columns: 4;
+    column-gap: $column-gap;
+    row-gap: $row-gap;
+  }
```

```suggestion
  column-gap: $column-gap;
  row-gap: $row-gap;
    
  @include onPhone {
    --columns: 2;
  }

  @include onTablet {
    --columns: 3;
  }

  @include onDesctop {
    --columns: 4;
  }
```
- Gaps are the same in all cases, so you don't need to specify for all media.
- You can use the shorthand property `gap` as well. https://developer.mozilla.org/ru/docs/Web/CSS/gap